src/unity/lib/unity_sarray.cpp

```diff
@@ -2836,79 +2836,17 @@ std::shared_ptr<model_base> unity_sarray::plot(const std::string& path_to_client
     log_and_throw("Nothing to show; SArray is empty.");
   }
 
-  std::shared_ptr<unity_sarray> self = \
-             std::make_shared<unity_sarray>(*this);
+  std::shared_ptr<unity_sarray> self = std::make_shared<unity_sarray>(*this);
+  gl_sarray sa = gl_sarray(self);
```

Comment:
I think you can do `gl_sarray sa(self)` to use copy constructor instead of copy assign operator (style preference, I think -- I think any efficiency difference is likely to get optimized away).